apps/prairielearn/src/server.js

```diff
@@ -150,22 +150,30 @@ module.exports.initExpress = function () {
   });
 
   app.use(
-    session({
+    createSessionMiddleware({
       secret: config.secretKey,
-      store: new SessionStore({
-        expireSeconds: config.sessionStoreExpireSeconds,
-      }),
-      resave: false,
-      saveUninitialized: true,
+      store: new NewSessionStore(),
       cookie: {
         maxAge: config.sessionStoreExpireSeconds * 1000,
-        httpOnly: true,
-        secure: 'auto', // uses Express "trust proxy"
+        name: 'prairielearn_session',
+        secure: 'auto', // uses Express "trust proxy" setting
         sameSite: config.sessionCookieSameSite,
       },
     }),
   );
 
+  app.use((req, res, next) => {
+    // If the session is going to expire in the near future, we'll extend it
+    // automatically for the user.
+    //
+    // TODO: make this configurable?
```

Comment:
👀 should this period be configurable?

Reply:
Probably? At some point we might need to be able to set it on a per-institution basis, but I'd defer all of this for as long as possible. I'm fine hard coding it for now.

Reply:
Sounds good, I'll leave it hardcoded for now.